### ISSUE TITLE
CoreOS-Beta-695.0.0 breaks etcd2 proxy deployment

### ISSUE THREAD

**Issue by arjanschaaf** [coreos/bugs#384]
My servers running on the beta channel upgraded from 681.0.0 to 695.0.0.
We have a dedicated etcd2 cluster with three nodes, the other CoreOS servers run etcd2 in proxy mode, using the following cloudinit configuration:

```
  etcd2:
    proxy: on
    listen-client-urls: 'http://127.0.0.1:2379'
    initial-cluster: 'node-1=http://<ip-address>:2380,node-2=http://<ip-address>:2380,node-3=http://<ip-address>:2380'
```

this setup runs fine on 681.0.0 and results in the following logs:

```
Jun 12 11:07:46 proxy-node-1 systemd[1]: Starting etcd2...
Jun 12 11:07:47 proxy-node-1 etcd2[657]: 2015/06/12 11:07:47 proxy: using peer urls [http://<ip-address>:2380 http://<ip-address>:2380 http://<ip-address>:2380]
Jun 12 11:07:47 proxy-node-1 etcd2[657]: 2015/06/12 11:07:47 proxy: listening for client requests on 127.0.0.1:2379
```

running exactly the same cloudinit configuration on 695.0.0 results in a etcd2 service that won't start and the following log data:

```
Jun 12 11:20:42 proxy-node-1 systemd[1]: Starting etcd2...
Jun 12 11:20:43 proxy-node-1 etcd2[660]: 2015/06/12 11:20:43 error verifying flags, -advertise-client-urls is required when -listen-client-urls is set explicitly. See 'etcd -help'.
Jun 12 11:20:43 proxy-node-1 etcd2[660]: 2015/06/12 11:20:43 When listening on specific address(es), this etcd process must advertise accessible url(s) to each connected client.
Jun 12 11:20:43 proxy-node-1 systemd[1]: etcd2.service: main process exited, code=exited, status=1/FAILURE
Jun 12 11:20:43 proxy-node-1 systemd[1]: Unit etcd2.service entered failed state.
Jun 12 11:20:43 proxy-node-1 systemd[1]: etcd2.service failed.
```
Almost as if the **proxy: on** in the configuration is completely ignored?

**Comment by xiang90** [comment]
This is fixed by https://github.com/coreos/etcd/pull/2951

**Comment by crawford** [comment]
This has been fixed in 695.2.0 (the current Beta).
